### PR TITLE
predict on dataframes with missing rows

### PR DESCRIPTION
**Pull request by matthieugomez**
`predict(mm::DataFrameRegressionModel, df::AbstractDataFrame)` currently returns a vector of length lower than `size(df, 1)` when there are missing regressors in the dataframe `df`,

I think it should always return a vector of length equal to `size(df, 1)` (potentially with missing values), so that it remains aligned with the initial dataframe, and so that one can  do
```
df[:yp] = predict(mm, df)
```

I know missing values are currently being redone so this pull request won't impact anything - but this is a general discussion and rewriting the current predict function makes my point clearer.

### CODE REVIEW

File: test/statsmodel.jl [hunk 1]
```diff
@@ -47,6 +47,10 @@ mm = ModelMatrix(ModelFrame(f, d))
 ## new data from DataFrame (via ModelMatrix)
 @test predict(m, d) == predict(m, mm.m)
 
+d2 = deepcopy(d)
+d2[3, :x1] = NA
+@test length(predict(m, d2)) = 4
```
Comment (kleinschmidt) [inline test/statsmodel.jl]: Should be `==`?

Comment (matthieugomez) [inline test/statsmodel.jl]: Yep.

**Comment by kleinschmidt** [comment]
This looks good to me.  The failing tests are just because of the missing `::Colon` indexing support which is fixed by #826.

**Comment by johnmyleswhite** [comment]
Seems like an improvement to me.

**Comment by garborg** [comment]
Looks like the nightlies are fine on Travis now. Merging.

**Pull request merged**